swap_meet/vendor.py

```diff
@@ -1,2 +1,207 @@
+        Parameters:
+            inventory - represents all of the objects of the classs Item
+              that self has and is willing to swap with other vendors
+        '''
+
+        #If no argument is passed for inventory it will initialize as an empty list
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, new_item):
+
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, old_item):
+        '''
+        input:
+            self -
+            old_item - object from the Item class that self will remove from its inventory list 
+        output:
+            old_item - if old_item is found in the inventory list it will be removed and returned 
+                side effect- inventory will have the old_item removed 
+            False - if old_item is not found inventory will not change
+        '''
+
+        if old_item in self.inventory:
+
+            self.inventory.remove(old_item)
+            return old_item
+
+        else:
+            return False
+        
+    
+    def get_by_id(self, items_id):
+        '''
+        Finds the item in self.inventory with the id = items_id
+        '''
+
+        for item in self.inventory:
+            if items_id == item.id: 
+                return item
+    
+
+    
+    def swap_items(self, other_vendor, my_item, others_item):
+        '''
+    input: 
+        self -
+        other_vendor - object of the same class vendor that self wants to trade with, 
+        my_item - Item self wants to swap, 
+        others_item- item other_vendor wants to swap
+    output: 
+        True - if self and othr_vendor have items that the other wants to swap
+            Side effect- the inventory parameter for self and other_vendor is modified by having the new item and
+                removing the old item
+        False - if self or other_vendor does not have its corrsponding item in their inventory parameter
+        '''
+
+        #Check each of the vendors has the correct item to swap 
+
+        if my_item in self.inventory and others_item in other_vendor.inventory:
+            
+            self.inventory.remove(my_item)
+            other_vendor.inventory.remove(others_item)
+            
+            self.inventory.append(others_item)
+            other_vendor.inventory.append(my_item)
+            
+            return True 
+        
+        # Return False if one of the vendors does not have the item
+        return False
+
+
+    
+    def swap_first_item(self, other_vendor):
+        '''
+        input:
+            self
+            other_vendor - another object of the same class (vendor)
+        output: 
+            False - If either of the vendors have empty inventories 
+            True - If both of the vendors have at least one item in their inventories 
+                and a swap is done 
+        '''
+
+        if len(self.inventory) and len(other_vendor.inventory):
```

Comment:
This could be a guard clause, too. You could write:

```python
        if not self.inventory or not other_vendor.inventory:
            return False
```